### ISSUE TITLE
Support for Jupyter notebooks

### ISSUE THREAD

**Issue by ahuang11** [TexteaInc/funix#86]
Wondering if it's possible to run in notebook, or at least indirectly.

In Panel, there's `panel_obj.show()` which opens a new browser.

**Comment by forrestbao** [comment]
Thanks @ahuang11 for this suggestion. We heard the same feature request from many others at PyCon US 2024. 

What's your suggestion on how a user can trigger the code-to-app conversion? 
As you know, Funix does not require the user to modify the code. So how can Jupyter know when is the time to map the user code to a GUI app? 
By adding a magic command and display the app when pressing Ctrl/Cmd + Enter? 
Or, by using a new button on Jupyter?

**Comment by ahuang11** [comment]
A decorator probably?

```
@funix
def ...
```

Or magic command works too I guess, but I like decorator
```
%funix

def ...
```

**Comment by ahuang11** [comment]
Perhaps the most Jupyter way of doing things is if the function is on the last line. If it is, display it.

**Comment by forrestbao** [comment]
Thanks. We will support at least one way to do it first. Then we can support more.

**Comment by Yazawazi** [comment]
On the develop branch, there is a way to run Funix indirectly in Jupyter:

```python
from funix import funix


@funix()
def hello(a: str) -> str:
    return f"Hello {a}!"
```

This will directly open a new Flask application and show this by iframe.

<img width="1744" alt="Image" src="https://github.com/TexteaInc/funix/assets/47273265/8b1997d9-82c7-470f-8fb1-edd2b5dc62fd">

In the future, I will try to communicate with the kernel for calling functions.

The following is a reference steps for using funix with the develop branch:

1. `git clone https://github.com/TexteaInc/funix`
2. `cd funix`
3. `git checkout develop`
4. `pip install -e .`

**Comment by ahuang11** [comment]
Cool, quick update!

I'm wondering what if it's possible to import funix once at an upper cell, and all subsequent cells will automatically show iframe if there's the function variable at the end?

Cell 1
```python
import funix
```

Cell 2
```python
def hello(a: str) -> str:
    return f"Hello {a}!"

hello
```

**Comment by Yazawazi** [comment]
I'll try to work on this idea 👀

**Comment by forrestbao** [comment]
I am not sure about this idea below because it conflicts with expectations per Jupyter convention. 
A user's expectation for the last line, `hello` is printing the variable `hello`, which here will be the location in the RAM of the function. 

What do you guys think? 

> Cool, quick update!
> 
> I'm wondering if it's possible to, separately, import funix.notebook once at an upper cell, and all subsequent cells will automatically show iframe if there's the function variable at the end?
> 
> Cell 1
> 
> ```python
> import funix.notebook
> ```
> 
> Cell 2
> 
> ```python
> def hello(a: str) -> str:
>     return f"Hello {a}!"
> 
> hello
> ```

**Comment by ahuang11** [comment]
> which here will be the location in the RAM of the function.

I personally have never utilized the location in the RAM of the function. Also, if they really need it, `print(hello)` would return that.

**Comment by forrestbao** [comment]
The expected behavior is as follows: 
```
In [1]: def hello(a: str) -> str:
   ...:     return "Hello, " + a + "!"
   ...: 

In [2]: hello
Out[2]: <function __main__.hello(a: str) -> str>

In [3]: print(hello)
<function hello at 0x70565a7a2320>
```

**Comment by ahuang11** [comment]
Yes, I think `display(hello)` results in `<function __main__.hello(a: str) -> str>`

<img width="310" alt="image" src="https://github.com/TexteaInc/funix/assets/15331990/244fc7a4-4215-4d61-9763-e2ca85813e6d">
